pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -275,7 +275,10 @@ func (r *ReconcileWindowsMachineConfig) addWorkerNode() (types.WindowsVM, error)
 	}
 
 	log.V(1).Info("configuring the Windows VM", "ID", vm.GetCredentials().GetInstanceId())
-	nc := nodeconfig.NewNodeConfig(r.k8sclientset, vm)
+	nc, err := nodeconfig.NewNodeConfig(r.k8sclientset, vm)
+	if err != nil {
```

Comment:
Please use `if err := nodeconfig.NewNodeConfig(r.k8sclientset, vm); err != nil` pattern.

Reply:
We cannot use it as nc's scope will be reduced to if block then.